code/web/src/modules/user/EditProfile.js

```diff
@@ -0,0 +1,125 @@
+import React, { useState } from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import Button from '../../ui/button'
+import { Grid, GridCell } from '../../ui/grid'
+import { H3 } from '../../ui/typography'
+import { grey} from '../../ui/common/colors'
+import { updateProfile } from './api/actions'
+
+
+
+// class EditProfile extends React.Component {
+const EditProfile = (props) => {
+
+    const fileInput = React.createRef()
+
+    const [ name, updateName ] = useState()
+    const [ email, updateEmail ] = useState()
+    const [ shippingAddress, updateShippingAddress ] = useState(
+    )
+    const [ description, updateDescription ] = useState()
+    // const [ image, updateImage ] = useState()
+
+    // handle picture upload
+    // const handleSubmit = e => {
+    //     e.preventDefault()
+    //     const pic = fileInput.current.files[0]
+    //     // console.log(image)
+    //     updateImage(pic)
+    // }
+    
+    const handleClick = async () => {
+        console.log(props.user)
+        const updatedUser = {
+            // id: props.user.id,
+            name,
+            email,
+            shippingAddress,
+            description
+        }
+        await props.updateProfile(updatedUser)
+        .then(data => console.log(data))
+        .catch(err => console.error(err))
+          
+
+        clearInputs()
+    }
+
+    const clearInputs = () => {
+        updateName('')
+        updateEmail('')
+        updateShippingAddress('')
+        updateDescription('')
+    }
```

Comment:
This clearInputs is a little redundant isn't it? If the user is successfully updated - we'd want to redirect them back to the user/profile path

Reply:
It's always a good idea to clear inputs on submit